### PR TITLE
Add Rust helpers

### PR DESCRIPTION
**Pull request by jackcmay**
#### Summary of Changes

- Adding sol_panic() function that causes an explicit panic helper that abort the program execution
- Added helper to catch calls to abort() which the compiler is inserting for unsupported operations

Fixes #

### CODE REVIEW

File: programs/native/bpf_loader/src/lib.rs [hunk 1]
```diff
@@ -29,6 +29,53 @@ fn dump_program(key: &Pubkey, prog: &[u8]) {
     }
 }
 
+#[allow(unused_variables)]
```
Comment (garious) [inline programs/native/bpf_loader/src/lib.rs]: Put underscores in front of the unused parameters instead?

File: programs/native/bpf_loader/src/lib.rs [hunk 2]
```diff
@@ -29,6 +29,53 @@ fn dump_program(key: &Pubkey, prog: &[u8]) {
...
+
+#[allow(unused_variables)]
+pub fn helper_sol_panic(
+    unused1: u64,
+    unused2: u64,
+    unused3: u64,
+    unused4: u64,
+    unused5: u64,
+) -> u64 {
+    // Never called because its verify function always returns an error
```
Comment (garious) [inline programs/native/bpf_loader/src/lib.rs]: it's

File: programs/native/bpf_loader/src/lib.rs (hunk 1 above)
Comment (jackcmay) [reply programs/native/bpf_loader/src/lib.rs]: I'm not partial, just following an example that Vines set, underlines actually make for more compact code, maybe change later.

File: programs/native/bpf_loader/src/lib.rs (hunk 2 above)
Comment (jackcmay) [reply programs/native/bpf_loader/src/lib.rs]: I want the possessive

Comment (garious) [reply programs/native/bpf_loader/src/lib.rs]: got it. I read it as "it's a verify function"

File: programs/native/bpf_loader/src/lib.rs (hunk 1 above)
Comment (garious) [reply programs/native/bpf_loader/src/lib.rs]: ok

Comment (garious) [reply programs/native/bpf_loader/src/lib.rs]: the underscores would ensure you're notified when one of the remaining parameters is accidentally unused (typically as a result of shadowing)

Comment (jackcmay) [reply programs/native/bpf_loader/src/lib.rs]: agreed, ok I'll change as part of this pr

**Pull request merged**